src/java/org/apache/cassandra/tcm/RemoteProcessor.java

```diff
@@ -148,6 +148,30 @@ public ClusterMetadata fetchLogAndWait(Epoch waitFor, Retry.Deadline retryPolicy
         }
     }
 
+    @Override
+    public LogState reconstruct(Epoch lowEpoch, Epoch highEpoch, Retry.Deadline retryPolicy)
+    {
+        try
+        {
+            Promise<LogState> request = new AsyncPromise<>();
+            List<InetAddressAndPort> candidates = new ArrayList<>(log.metadata().fullCMSMembers());
+            sendWithCallbackAsync(request,
+                                  Verb.TCM_RECONSTRUCT_EPOCH_REQ,
+                                  new ReconstructLogState(lowEpoch, highEpoch),
+                                  new CandidateIterator(candidates),
+                                  new Retry.Backoff(TCMMetrics.instance.fetchLogRetries));
+            return request.get(retryPolicy.remainingNanos(), TimeUnit.NANOSECONDS);
```

Comment:
Should we store the returned LogState in the local tables?

Reply:
From Accord usage, this was a single-shot feature. I am a bit concerned about storing the state in the local table because it'll require a pretty thorough set of tests, since technically _any_ of the epochs can be a snapshot, and we can not / should not leave any gaps. Does not mean we can not do this of course.

Reply:
I don't know how accord will actually use this, but I imagine this to be quite "bursty" - like right after we join, there would be many transactions using older epochs that need reconstruction, so a debounce + store the result locally would reduce the load on the CMS

But maybe thats not how it will be used?

What does "single-shot feature" mean here?

Reply:
I just meant that it will be used during bootstraps in order to let the bootstrapping node reconstruct in-progress epochs, and there usually should be only few of them. If you expand a larger cluster, you may get as many requests as bootstrapping nodes, but I would argue this is still not really bursty, this is a very small number of requests proportional to the size of the cluster.